.github/workflows/release.yaml

```diff
@@ -0,0 +1,91 @@
+# When pushing to main, we run checks and then release.
+
+name: Check and Release
+on:
+  push:
+    branches:
+    - main
+permissions:
+  contents: read
+env:
+  ARTIFACT_NAME: artifact-ubuntu-latest-3.10
+
+jobs:
+  build:
+    uses: ./.github/workflows/build.yaml
+    permissions:
+      contents: write # To create the release tag.
+    secrets: inherit
+
+  provenance:
+    needs: build
+    permissions:
+      actions: read   # To read the workflow path.
+      id-token: write # To sign the provenance.
+      contents: write # To add assets to a release.
+    # The generator should be referenced with a semantic version.
+    # The build will fail if we reference it using the commit sha.
```

Comment:
Dang! Is that a Github problem?

Reply:
> 2\. Don’t create the venv and simply operate in the Action container’s global Python installation

I'll go with this option.

Reply:
> 

The `Makefile` expects `.venv/upgraded-on`, so we need to investigate this in a separate PR. I left a comment in `build.yaml`.

Reply:
Issue https://github.com/jenstroeger/python-package-template/issues/267.